### ISSUE TITLE
Galaxy minifyStdCSS_plugin cannot read getContentAsString of undefined

### ISSUE THREAD

**Issue by dimadev01** [meteor/meteor#10112]
Hi!

I have a problem with the Meteor 1.7.1.beta-* when trying to use meteor-now or in this case deploying to Galaxy servers.

This is the error when trying to deploy (locally it works perfectly)

```
Talking to Galaxy servers at https://galaxy.meteor.com
Preparing to deploy your app...
Node#moveTo was deprecated. Use Container#append.

Errors prevented deploying:
While minifying app stylesheet:
packages/minifyStdCSS_plugin.js:130:34: Cannot read property 'getContentsAsString' of undefined
at packages/minifyStdCSS_plugin.js:130:34
at Array.map (<anonymous>)
at packages/minifyStdCSS_plugin.js:129:32
at CssToolsMinifier.processFilesForBundle (packages/minifyStdCSS_plugin.js:33:16)


While minifying app stylesheet:
packages/minifyStdCSS_plugin.js:130:34: Cannot read property 'getContentsAsString' of undefined
at packages/minifyStdCSS_plugin.js:130:34
at Array.map (<anonymous>)
at packages/minifyStdCSS_plugin.js:129:32
at CssToolsMinifier.processFilesForBundle (packages/minifyStdCSS_plugin.js:33:16)
```

Version where it worked: 1.7.0.3
Version where it does not work: 1.7.1.beta-*
My OS is MacOS
The expected behavior is: the app being deployed successfully
The actual behavior is an exception when deploying the app

Thanks!

**Comment by benjamn** [comment]
@KoenLav Can you reproduce this locally by running `meteor --production`?

**Comment by KoenLav** [comment]
@benjamn yep, same problem locally (Windows 10) running meteor --production.

**Comment by benjamn** [comment]
Ok, so I think the best way to debug this is to copy the `standard-minifier-css` package directory into your local `packages` directory, then add some `if (! file) debugger` statements before the lines with `file.getContentsAsString()`, and then run
```sh
$env.TOOL_NODE_FLAGS = "--inspect-brk"
meteor --production
```
The `$env` thing assumes you're using PowerShell, but you can adjust for whatever shell you're using (I'm just trying to set the `TOOL_NODE_FLAGS` environment variable).

**Comment by KoenLav** [comment]
I think this is related to:

https://github.com/mozfet/meteor-autoform-materialize/issues/43

as I'm also seeing these messages:

   error: Scss compiler error: Error: File to import: {}/node_modules/materialize-css/sass/components/_color-variables.scss not found in file:
   C:/Users/koenl/Projects/MrWinston/waiter/{mozfet:autoform-materialize}/style.scss
   on line 1 of {mozfet:autoform-materialize}/style.scss
   >> @import "{}/node_modules/materialize-css/sass/components/_color-variables.sc

**Comment by KoenLav** [comment]
The problem is resolved when I create a symbolic link at imports/materialize-css to node_modules/materialize-css.

**Comment by KoenLav** [comment]
Well, resolved on my local machine, but when committing the symbolic link to Git and running another build on CircleCI I get the same error.

Probably caused by Git not understanding the symbolic link created on Windows correctly?

**Comment by smeijer** [comment]
For what it's worth, I'm experiencing the same `While minifying app stylesheet ... Cannot read property 'getContentsAsString' of undefined` error.

I removed all `.map` files in the project to be sure it's not a problem with existing `.map` files in `node_modules` (as suggested in https://github.com/meteor/meteor/issues/9610#issuecomment-365675812), but that didn't help either.

**Comment by mjjansen** [comment]
same...
```
While minifying app stylesheet:
   packages/minifyStdCSS_plugin.js:130:34: Cannot read
   property 'getContentsAsString' of undefined
   at packages/minifyStdCSS_plugin.js:130:34
   at Array.map (<anonymous>)
   at packages/minifyStdCSS_plugin.js:129:32
   at CssToolsMinifier.processFilesForBundle
   (packages/minifyStdCSS_plugin.js:33:16)


   While minifying app stylesheet:
   packages/minifyStdCSS_plugin.js:130:34: Cannot read
   property 'getContentsAsString' of undefined
   at packages/minifyStdCSS_plugin.js:130:34
   at Array.map (<anonymous>)
   at packages/minifyStdCSS_plugin.js:129:32
   at CssToolsMinifier.processFilesForBundle
   (packages/minifyStdCSS_plugin.js:33:16)
```

**Comment by smeijer** [comment]
The debugger statement has some other issues (also when I change port):

```
Debugger attached.
Starting inspector on 127.0.0.1:9229 failed: address already in use
Waiting for the debugger to disconnect...
```

So I went ahead with `console.log` statements as well. And, it seem like we are having a same issue here:

```js
if (!originals[filename]) {
  console.log('going to read failing', filename);
}

return originals[filename].getContentsAsString();
```

```bash
going to read failing app/node_modules/react-virtualized/source/styles.css
```

**Comment by benjamn** [comment]
Does that file (`node_modules/react-virtualized/source/styles.css`) exist?

**Comment by smeijer** [comment]
And a repo:

```bash
git clone git@github.com:smeijer/meteor-10112.git
meteor npm install
meteor --production
```

```bash
=> Started proxy.
=> Started MongoDB.
=> Errors prevented startup:

   While minifying app stylesheet:
   packages/minifyStdCSS_plugin.js:130:34: Cannot read property 'getContentsAsString' of undefined
   at packages/minifyStdCSS_plugin.js:130:34
   at Array.map (<anonymous>)
   at packages/minifyStdCSS_plugin.js:129:32
   at CssToolsMinifier.processFilesForBundle (packages/minifyStdCSS_plugin.js:33:16)


   While minifying app stylesheet:
   packages/minifyStdCSS_plugin.js:130:34: Cannot read property 'getContentsAsString' of undefined
   at packages/minifyStdCSS_plugin.js:130:34
   at Array.map (<anonymous>)
   at packages/minifyStdCSS_plugin.js:129:32
   at CssToolsMinifier.processFilesForBundle (packages/minifyStdCSS_plugin.js:33:16)


=> Your application has errors. Waiting for file change.
```

**Comment by smeijer** [comment]
@benjamn, no it doesn't. But I also don't include it anywhere. The correct path would be /node_modules/react-virtualized/style.css.

Only thing I did in this repo was:

```
meteor create meteor-10112
cd meteor-10112
npm install react-virtualized

# add "import 'react-virtualized/styles.css';" to client/main.js

meteor run --production
```

**Comment by mjjansen** [comment]
> Does that file (`node_modules/react-virtualized/source/styles.css`) exist?

I think it is missing.  I see it under `node_modules/react-virtualized/styles.css`, not `../source/styles.css`

**Comment by mjjansen** [comment]
so this fixed it:
```
cd node_modules/react-virtualized
mkdir source
ln -s styles.css source/styles.css
```

**Comment by smeijer** [comment]
I tried copying the file on Windows, but didn't work:

```bash
λ mkdir node_modules\react-virtualized\source
λ cp node_modules\react-virtualized\styles.css node_modules\react-virtualized\source\
```

**Comment by benjamn** [comment]
I'll have a fix for this momentarily, but I wanted to explain the problem briefly: the `react-virtualized/styles.css` file, which is what's being imported, includes an inline `/*# sourceMappingURL=...` source map, which refers to the original source file `react-virtualized/source/styles.scss`. When Meteor processes the source maps, it tries to go all the way back to the original source (as long as it has source maps), but `react-virtualized/source/styles.scss` is not one of the files that's being minified (since it wasn't explicitly imported)—it's just a file that happens to be mentioned by a source map.

In other words, we can fix this by
* actually trying to read the unknown file from disk, rather than using `getContentsAsString`,
* skipping unrecognized files and removing them from `stringifiedCss.map.sources`, or
* stripping any stray source maps from `mergedCssAst` before we call `CssTools.stringifyCss(mergedCssAst, ...)`

**Comment by smeijer** [comment]
I think 'stripping before we call `CssTools.stringifyCss` sounds like the best solution. But I'll leave that call up to you :)

Nice to read that we can have a fix so soon. Awesome work! 👍

**Comment by KoenLav** [comment]
@benjamn do you also have reason to believe this PR will fix the issue we are experiencing?

Which seemed to be resolved using symbolic links, but not on CircleCI yet?

**Comment by benjamn** [comment]
Please try running `meteor update standard-minifier-css` to get version 1.5.1, which should fix these problems.

**Comment by mjjansen** [comment]
well done and extremely fast turnaround!

**Comment by KoenLav** [comment]
@benjamn I can confirm this also fixes our specific problem (without symlinking materialize-css in the imports directory).
